clang c++23 std::format compilation error

```
option(CAF_USE_STD_FORMAT "Enable std::format support" ON)
cmake -DCMAKE_CXX_FLAGS="-stdlib=libc++" -B build
```
```
clang version 19.1.5
cmake version 3.31.0
ninja version 1.11.1
debian 12
```
error:
```
/actor-framework-main/libcaf_core/caf/detail/format.test.cpp:30:3: error: static assertion failed due to requirement 'std::is_same_v<std::vector<int, std::allocator<int>> &&, std::string>'
   30 |   std::is_same_v<decltype(fmt_fwd(std::declval<std::vector<int>>())),
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                  std::string>);
      |                  ~~~~~~~~~~~~
1 error generated.
[255/667] Building CXX object libcaf_core/CMakeFiles/caf-core-test.dir/caf/detail/parse.test.cpp.o
```

--- a/libcaf_core/caf/detail/format.test.cpp
+++ b/libcaf_core/caf/detail/format.test.cpp
@@ -27,8 +27,7 @@ static_assert(
   std::is_same_v<decltype(fmt_fwd(std::declval<const int&>())), const int&>);
 // Types without a formatter overload are forwarded as std::string.
 static_assert(
-  std::is_same_v<decltype(fmt_fwd(std::declval<std::vector<int>>())),
-                 std::string>);
+  std::is_same_v<decltype(fmt_fwd(std::declval<caf::uri>())), std::string>);
 
 #endif
 

